packages/editor/src/components/provider/index.js

```diff
@@ -140,21 +136,6 @@ class EditorProvider extends Component {
 		if ( this.props.settings !== prevProps.settings ) {
 			this.props.updateEditorSettings( this.props.settings );
 		}
-
```

Comment:
As noted in the description:

> I removed the logic which uninstalls the block/plugin if it isn't used in the post. I can bring it back once we agree that the general direction is sound.

Reply:
With the current approach, you can wrap the registered UI Plugin with such component. This way it would work for all types of editors: `edit-post`, `edit-widgets` and other contexts we add in the future.

Reply:
Added a note in #16524.

It was already listed as an action item in #17440.